ui/src/cloud/reducers/limits.ts

```diff
@@ -66,33 +66,50 @@ export const limitsReducer = (
       case ActionTypes.SetLimits: {
         const {limits} = action.payload
 
-        const {maxBuckets, maxRetentionDuration} = limits.bucket
-        const {maxDashboards} = limits.dashboard
-        const {maxTasks} = limits.task
-        const {maxChecks} = limits.check
-        const {
-          maxNotifications,
-          blockedNotificationRules,
-        } = limits.notificationRule
-        const {blockedNotificationEndpoints} = limits.notificationEndpoint
-
-        const {readKBs, writeKBs, cardinality} = limits.rate
-
-        draftState.buckets.maxAllowed = maxBuckets
-        draftState.buckets.maxRetentionSeconds = maxRetentionDuration / 1e9
-        draftState.dashboards.maxAllowed = maxDashboards
-        draftState.tasks.maxAllowed = maxTasks
-        draftState.checks.maxAllowed = maxChecks
-        draftState.rules.maxAllowed = maxNotifications
-        draftState.rules.blocked = blockedNotificationRules
-          .split(',')
-          .map(r => r.trim())
-        draftState.endpoints.blocked = blockedNotificationEndpoints
-          .split(',')
-          .map(r => r.trim())
-        draftState.rate.readKBs.maxAllowed = readKBs
-        draftState.rate.writeKBs.maxAllowed = writeKBs
-        draftState.rate.cardinality.maxAllowed = cardinality
+        if (limits.bucket) {
```

Comment:
i couldn't find what the shape of `limits.bucket` or `limits.dashboard` etc are. Only thing that stands out to me is this truthy check might be wonky depending on the shape of `lmiits.bucket`

e.g. if `limits.bucket` is `undefined`, we're good. if `limits.bucket` is `{}`, this check will pass. I just don't have context into this, figured I'd mention it.

Reply:
it also started begging the question as to why v2private isn't being routed through a swagger defs so that we could normalize this ish and guarantee a shape. i figure that pathway will be ironed out next quarter with the quartz work though and we can look at it then.